capnpc/rust.capnp

```diff
@@ -26,3 +26,20 @@ annotation parentModule @0xabee386cd1450364 (file) :Text;
 #      }
 #    }
 #  }
+
+annotation getOption @0xabfef22c4ee1964e (field) :Void;
+# Whether the generated code should return an Option. Supported on Text, Data,
+# List, and struct fields.
```

Comment:
Is there any reason for this to not work on any pointer type?

Reply:
I forgot :). Went back to the capnp docs and I think I'm just missing AnyPointer and interfaces.

Just implemented AnyPointer support.

I just spent about an hour hitting my head against a wall trying to understand capnp interfaces, which I've never used before. I couldn't figure out how to do something as simple as set a field in a message to an empty interface. There are some places in the current code that look buggy (for ex I think [this](https://github.com/danielzfranklin/capnproto-rust/blob/f82031359ebf4e8824d09d416f97d85890de74b1/capnpc/src/codegen.rs#LL664C40-L664C41) breaks interfaces nested in things). I don't see any examples or tests for interfaces in this repo.

Do you know any examples of code using interfaces I could read? I'm considering leaving them out of this PR.